ui/src/components/NodePageMetricsTab.js

```diff
@@ -87,98 +85,98 @@ const NodePageMetricsTab = (props) => {
       sampleFrequency,
     );
 
-  const cpuUsageOperated = operateMetricRawData(nodeStats?.cpuUsage[0]?.values);
-  const systemLoadOperated = operateMetricRawData(
-    nodeStats?.systemLoad[0]?.values,
-  );
-  const memoryOperated = operateMetricRawData(nodeStats?.memory[0]?.values);
-  const iopsWriteOperated = operateMetricRawData(
-    nodeStats?.iopsWrite[0]?.values,
-  );
-  const iopsReadOperated = operateMetricRawData(nodeStats?.iopsRead[0]?.values);
-  const controlPlaneNetworkBandwidthInOperated = operateMetricRawData(
-    nodeStats?.controlPlaneNetworkBandwidthIn[0]?.values,
-  );
-  const controlPlaneNetworkBandwidthOutOperated = operateMetricRawData(
-    nodeStats?.controlPlaneNetworkBandwidthOut[0]?.values,
-  );
-  const workloadPlaneNetworkBandwidthInOperated = operateMetricRawData(
-    nodeStats?.workloadPlaneNetworkBandwidthIn[0]?.values,
-  );
-  const workloadPlaneNetworkBandwidthOutOperated = operateMetricRawData(
-    nodeStats?.workloadPlaneNetworkBandwidthOut[0]?.values,
-  );
-
-  const cpuUsageData = cpuUsageOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
-    };
-  });
-  const systemLoadData = systemLoadOperated?.map((slot) => {
+  const nodeStatsOperated = Object.keys(nodeStats).map((metricName) => {
     return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
-    };
-  });
-  const memoryData = memoryOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
+      metrics: metricName,
+      data: operateMetricRawData(nodeStats[metricName][0]?.values),
     };
   });
 
-  const nodeIOPSWriteData = iopsWriteOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      write: slot[1],
-      type: 'write',
-    };
-  });
-  const nodeIOPSReadData = iopsReadOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      read: slot[1],
-      type: 'read',
-    };
-  });
-  const nodeControlPlaneNetworkBandwidthInData = controlPlaneNetworkBandwidthInOperated?.map(
-    (slot) => {
+  const cpuUsageData = nodeStatsOperated
+    ?.find((obj) => obj.metrics === 'cpuUsage')
+    ?.data?.map((slot) => {
+      return {
+        date: fromUnixTimestampToDate(slot[0]),
+        y: slot[1],
+      };
+    });
+
+  const systemLoadData = nodeStatsOperated
+    ?.find((obj) => obj.metrics === 'systemLoad')
+    ?.data?.map((slot) => {
+      return {
+        date: fromUnixTimestampToDate(slot[0]),
+        y: slot[1],
+      };
+    });
+
+  const memoryData = nodeStatsOperated
+    ?.find((obj) => obj.metrics === 'memory')
+    ?.data?.map((slot) => {
+      return {
+        date: fromUnixTimestampToDate(slot[0]),
+        y: slot[1],
+      };
+    });
+
+  const nodeIOPSWriteData = nodeStatsOperated
+    ?.find((obj) => obj.metrics === 'iopsRead')
```

Comment:
Looks like you switched them up, no?

```suggestion
    ?.find((obj) => obj.metrics === 'iopsWrite')
```